Makefile

```diff
@@ -272,20 +272,24 @@ manifests: YQ controller-gen ## Generate WebhookConfiguration, ClusterRole and C
 	$(YQ) -i 'del(.spec.versions[].schema.openAPIV3Schema.properties.spec.properties.consolePlugin.properties.advanced.properties.affinity.properties | .. | select(has("description")) | .description)' config/crd/bases/flows.netobserv.io_flowcollectors.yaml
 
 gencode: controller-gen ## Generate code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.
```

Comment:
Before you move forward, I suggest you add e2e tests to validate the metrics endpoint after the changes. Projects created with Kubebuilder's latest versions have comprehensive E2E tests, including code to validate the metrics endpoint. [Example](https://github.com/kubernetes-sigs/kubebuilder/tree/v4.3.1/testdata/project-v4/test/e2e). See: https://github.com/kubernetes-sigs/kubebuilder/blob/master/testdata/project-v4/test/e2e/e2e_test.go#L166-L235 (Note that the Prometheus part is not required)

Reply:
So, I'll need to dig deeper on how this is all designed, but maybe not in this PR. Those new tests seem quite different from the ones that we have, it might be a bigger refactoring (and writing tests to check metrics with our current framework, which uses kube REST API instead of `kubectl`, might be tricky)
(I'm not too worried about not having those tests right now, because that should be covered by QE automated tests - the operator generates dashboards from those metrics, which are tested)